src/App.js

```diff
@@ -0,0 +1,61 @@
+import React, { Component } from "react";
+import "./App.css";
+import Transactions from "./components/Transactions";
+import Operations from "./components/Operations";
+const axios = require('axios')
+class App extends Component {
+  constructor() {
+    super();
+    this.state = {
+      data: [],
+      balance: 0
+    };
+  }
+  componentDidMount = () => {
+   this.getDataFromDB()
+  }
+
+  getDataFromDB = async () =>{
+      let transactions = await axios.get("http://localhost:3003/transactions")
+      this.calcTotalBalance()
+      this.setState({
+        data : transactions.data
+      }, function(){
+        this.calcTotalBalance()
+      })
+     }
+
+  calcTotalBalance = () => {
+    let sum = this.state.data.map(a => a.amount).reduce((a, b) => a + b, 0)
+    this.setState({
+      balance : sum
+    })
+  }
+
+  deposit = async (amount, vendor, category, status) => {
+      let t1 = {amount, vendor , category, status}
+      await axios.post("http://localhost:3003/transaction", t1)
+      this.getDataFromDB()
+  }
+
+  withdraw = async (amount, vendor, category, status) => {
+    amount *= (-1)
+    let t1 = {amount, vendor , category, status}
+    await axios.post("http://localhost:3003/transaction", t1)
+    this.getDataFromDB()
+}
```

Comment:
the difference between `deposit` and `withdraw` is only one line... you could make one method that handle both options